src/apps/beefy/positions.ts

```diff
@@ -220,35 +220,41 @@ const beefyBaseVaultsPositions = async ({
 
   const userVaults: (BaseBeefyVault & { balance: bigint })[] = []
 
-  await Promise.all(
-    createBatches(vaults).map(async (batch) => {
-      if (batch.length === 0) {
-        return
-      }
-      const balances = await client.readContract({
-        abi: beefyV2AppMulticallAbi,
-        address: multicallAddress,
-        functionName: 'getTokenBalances',
-        args: [batch.map((vault) => vault.earnContractAddress), address],
-      })
-      for (let i = 0; i < balances.length; i++) {
-        if (balances[i] > 0) {
-          userVaults.push({
-            ...batch[i],
-            balance: balances[i],
-          })
+  if (address) {
+    await Promise.all(
+      createBatches(vaults).map(async (batch) => {
+        if (batch.length === 0) {
+          return
         }
-      }
-    }),
-  )
+        const balances = await client.readContract({
+          abi: beefyV2AppMulticallAbi,
+          address: multicallAddress,
+          functionName: 'getTokenBalances',
+          args: [batch.map((vault) => vault.earnContractAddress), address],
+        })
+        for (let i = 0; i < balances.length; i++) {
+          if (balances[i] > 0) {
+            userVaults.push({
+              ...batch[i],
+              balance: balances[i],
+            })
+          }
+        }
+      }),
+    )
+  } else {
+    userVaults.push(
+      ...vaults.map((vault) => ({ ...vault, balance: BigInt(0) })),
+    )
+  }
 
   if (!userVaults.length) {
     return []
   }
 
   const clmVaults = userVaults.filter((vault) => vault.type === 'cowcentrated')
   const info =
-    clmVaults.length === 0
+    clmVaults.length === 0 || !address // earn positions can't include clm vaults for now
```

Comment:
What's blocking including CLM vaults?

Reply:
I believe they all have multiple base tokens and are returned as contract positions, which is currently not supported in the wallet